sound/soc/sof/sof-audio.c

```diff
@@ -653,24 +653,37 @@ int sof_set_up_pipelines(struct device *dev)
 }
 
 /*
- * This function doesn't free widgets. It only resets the set up status for all routes and
- * use_count for all widgets.
+ * This function doesn't free widgets during suspend. It only resets the set up status for all
+ * routes and use_count for all widgets.
  */
-void sof_tear_down_pipelines(struct device *dev)
+int sof_tear_down_pipelines(struct device *dev, bool verify)
```

Comment:
Sorry, I am missing the concept on why we need a verify flag on teardown?

It's useful on setup to load all pipelines and check what the topology does. But on suspend I don't get what you are trying to do.

Reply:
The flag only tells you whether you need to tear down the pipelines or not. Previously we have only one case where we called tear_down_pipelines and that was during suspend. And during suspend we dont free anything, we just simply power down the DSP.

But now that we want to do a tplg verification during load, we should tear it down after the verification is complete. And when we tear the pipelines down during verification, we should actually free the widgets and so the verify flag needs to used for both set up and teardown

Reply:
ah ok, so you are also verifying the IPC free, but do we expect any errors, or is it just to go back to a blank slate?

Reply:
Yes, we do expect errors during verification and we return the error so the card wont be registered.